crates/remote_path/src/remote_path.rs

```diff
@@ -0,0 +1,71 @@
+use std::path::{Path, PathBuf};
+
+#[derive(Debug, Clone, Copy)]
+pub enum PathStyle {
+    Posix,
+    Windows,
+}
+
+impl PathStyle {
+    #[cfg(target_os = "windows")]
+    pub const fn current() -> Self {
+        PathStyle::Windows
+    }
+
+    #[cfg(not(target_os = "windows"))]
+    pub const fn current() -> Self {
+        PathStyle::Posix
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct RemotePathBuf {
```

Comment:
This is interesting, curious if the [typed_path](https://docs.rs/typed-path/latest/typed_path/#typed-paths) crate works for this? Or if it's easier to use our own?